crates/bevy_window/src/lib.rs

```diff
@@ -24,15 +24,34 @@ use bevy_app::prelude::*;
 use bevy_ecs::{event::Events, schedule::SystemLabel};
 
 pub struct WindowPlugin {
+    /// Whether to create a window when added.
+    ///
+    /// Note that if there are no windows, by default the App will exit,
+    /// due to [`exit_on_all_closed`].
     pub add_primary_window: bool,
-    pub exit_on_close: bool,
+    /// Whether to exit the app when there are no open windows.
+    /// If disabling this, ensure that you send the [`bevy_app::AppExit`]
+    /// event when the app should exit. If this does not occur, you will
+    /// create 'headless' processes (processes without windows), which may
+    /// surprise your users. It is recommended to leave this setting as `true`.
+    ///
+    /// If true, this plugin will add [`exit_on_all_closed`] to [`CoreStage::Update`].
+    pub exit_on_all_closed: bool,
+    /// Whether to close windows when they are requested to be closed (i.e.
+    /// when the close button is pressed)
+    ///
+    /// If true, this plugin will add [`close_when_requested`] to [`CoreStage::Update`].
+    /// If this system (or a replacement) is not running, the close button will have no effect.
+    /// This may surprise your users. It is recommended to leave this setting as `true`.
+    pub close_when_requested: bool,
```

Comment:
I'm not sure this field should exist. I agree with you that it's unintuitive behavior to set this to false; when would app developers want to do this?

Reply:
For example, if the user has unsaved changes, it's traditional to give an 'unsaved changes' warning window.

Reply:
Ah, got it. In that case, this should be exposed as a resource: that behavior needs to be able to be changed dynamically.

Reply:
Not really? If a developer wants this functionality, they can easily enough add it themselves. This is merely making sure the default behaviour is correct, and we give an escape hatch for when people need more complicated logic.

That is, I don't really want to encourage disabling this, but I want to make the option available.